apis/v1beta1/validation/gateway_test.go

```diff
@@ -247,6 +247,63 @@ func TestValidateGateway(t *testing.T) {
 			},
 			expectErrsOnFields: nil,
 		},
+		"ip address and hostname in addresses are valid": {
+			mutate: func(gw *gatewayv1b1.Gateway) {
+				gw.Spec.Addresses = []gatewayv1b1.GatewayAddress{
+					{
+						Type:  ptrTo(gatewayv1b1.IPAddressType),
+						Value: "1.2.3.4",
+					},
+					{
+						Type:  ptrTo(gatewayv1b1.HostnameAddressType),
+						Value: "foo.bar",
+					},
+				}
+			},
+			expectErrsOnFields: nil,
+		},
+		"ip address and hostname in addresses are invalid": {
+			mutate: func(gw *gatewayv1b1.Gateway) {
+				gw.Spec.Addresses = []gatewayv1b1.GatewayAddress{
+					{
+						Type:  ptrTo(gatewayv1b1.IPAddressType),
+						Value: "1.2.3.4:8080",
+					},
+					{
+						Type:  ptrTo(gatewayv1b1.HostnameAddressType),
+						Value: "*foo/bar",
+					},
+					{
+						Type:  ptrTo(gatewayv1b1.HostnameAddressType),
+						Value: "12:34:56::",
+					},
+				}
+			},
+			expectErrsOnFields: []string{"spec.addresses[0]", "spec.addresses[1]", "spec.addresses[2]"},
```

Comment:
While I suspect there may be some precedent somewhere else, I'm not a huge fan of this method of checking errors because it just checks _that_ there are errors and doesn't validate the consistency of the actual error produced. We should verify that the correct error type is produced (e.g. `Invalid`, `Duplicate`, e.t.c.) across all these tests, and also test the error text context. This will help to avoid future developers accidentally introducing subtle changes to error output on accident.

Reply:
not a fan of this method either, and I can improve that

Reply:
Thank you, very much appreciated :bow:

Reply:
hi, the changes has been made, PTAL